tests/formatter.rs

```diff
@@ -515,13 +515,26 @@ fn test_anon_lines() {
                 range: (0, 19),
             },
         },
+        DisplayLine::Source {
+            lineno: None,
+            inline_marks: vec![],
+            line: DisplaySourceLine::Empty,
```

Comment:
Should the case of `DisplaySourceLine::Empty` + `Some(lineno)` also be handled in some way? I didn't see any test cases for it, so I left that out for now.